### PR TITLE
Blog App: Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by forLoop94**
### The following changes were made:

-Fix n+1 problems

-Integration specs
      I used Capybara to write integration tests for each view in the project.
      
      User index page:
      
      I can see the username of all other users.
      I can see the profile picture for each user.
      I can see the number of posts each user has written.
      When I click on a user, I am redirected to that user's show page.

      user show page:
      
      I can see the user's profile picture.
      I can see the user's username.
      I can see the number of posts the user has written.
      I can see the user's bio.
      I can see the user's first 3 posts.
      I can see a button that lets me view all of a user's posts.
      When I click a user's post, it redirects me to that post's show page.
      When I click to see all posts, it redirects me to the user's post's index page.

      User post index page:
      
      I can see the user's profile picture.
      I can see the user's username.
      I can see the number of posts the user has written.
      I can see a post's title.
      I can see some of the post's body.
      I can see the first comments on a post.
      I can see how many comments a post has.
      I can see how many likes a post has.
      When I click on a post, it redirects me to that post's show page.

      Post show page:
      
      I can see the post's title.
      I can see who wrote the post.
      I can see how many comments it has.
      I can see how many likes it has.
      I can see the post body.
      I can see the username of each commentor.
      I can see the comment each commentor left.

       before n+1 query problem solving:
n+1_before

      after n+1 query problem solving:      
n+1_after

### CODE REVIEW

File: spec/system/user/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      expect(page).to have_content('My third post')
+      expect(page).to have_content('My fourth post')
+    end
+
+    it "When I click a user's post, it redirects me to that post's show page." do
+      visit user_path(@user1)
+      click_link("post_#{@posts[0].id}_link")
+      page.should have_button('Add comments')
+    end
+  end
```
Comment (anagudelogu) [inline spec/system/user/user_show_spec.rb]: - [ ] You did a great job working on this milestone! 💯 However, there's a missing test case:

image

Please consider adding it! 🤓 🔧

File: spec/system/post/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      visit user_posts_path(@user1)
+      expect(page).to have_content('Comments: 1 | Likes: 0')
+    end
+
+    it "When I click on a post, it redirects me to that post's show page." do
+      visit user_posts_path(@user1)
+      click_link(@posts[0].title.to_s)
+      expect(page).to have_button('Like')
+    end
+  end
```
Comment (anagudelogu) [inline spec/system/post/post_index_spec.rb:101]: - [ ] According to the requirements:

image

This test case is missing. Please consider including it! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @forLoop94 @Peter-Dumbari, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by forLoop94** [comment]
Missing test cases have been added. Thank you for the observation

**Review by mikethreels — APPROVED** [review]
Hi @forLoop94, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**